modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/pagemem/PagesWriteSpeedBasedThrottle.java

```diff
@@ -29,62 +30,26 @@
 /**
  * Throttles threads that generate dirty pages during ongoing checkpoint.
  * Designed to avoid zero dropdowns that can happen if checkpoint buffer is overflowed.
- * Uses average checkpoint write speed and moment speed of marking pages as dirty.<br>
+ * When a page is in checkpoint and Checkpoint Buffer is filled over 2/3, uses exponentially
```

Comment:
The overall logic looks like this:

```
if (isPageInCheckpoint && cpBufferIsInDangerZone()) {
    throttleExponentiallyToProtectCPBuffer();
} else {
    maybeApplySpeedBasedThrottlingToProtectCleanPages();
}
```

`isPageInCheckpoint` is a parameter of `onMarkDirty()` method.

'When a page is in checkpoint' means that `isPageInCheckpoint` is `true`. `onMarkDirty()` is also called from `PageMemoryImpl#allocate()` where `false` is passed as `isPageInCheckpoint`.

Would 'When the page at question is not included in the current checkpoint' instead of 'When a page is in checkpoint' be sufficient, what do you think?

Reply:
> Would 'When the page at question is not included in the current checkpoint' instead of 'When a page is in checkpoint' be sufficient, what do you think?

Yes, that's clear to me now

Reply:
Applied the change as 'When the page in question is not included in the current checkpoint' ('in question', not 'at question').